Docs-11375 add runtime update topic (#389)
mulesoft/docs-runtime-fabric: modules/ROOT/pages/runtime-patch-updates.adoc

= Mule Runtime Patch Updates for Runtime Fabric 

> The following provides important information about Mule runtime engine patch updates for Runtime Fabric.

Reviewer: First time in Level-1 topics is my own gen. rule:
```suggestion
The following provides important information about Mule runtime engine patch updates for Anypoint Runtime Fabric.
```

== Monthly Date-Patch Updates 

Once per month, MuleSoft releases a _date-patch update_, which includes backward-compatible bug fixes to address any issues discovered in Mule runtime engine.
The monthly date-patch update might include security patches if the scheduled date is within the security SLA.

The format of the version number for these updates is:

_Major_._Minor_._Patch_ - _Date_ 

Here's an example version number, including the patch date:

`4.3.0:20210101`

The monthly patches update applications only to the latest date-patch. They do not change the runtime version.

== Runtime Fabric Runtime Patch Update Schedule 

Runtime Fabric runs each application and gateway deployment on a separate Mule runtime instance. By managing the containerized Mule images used for these deployments, Runtime Fabric enables you to upgrade your Mule runtime version for each deployment on the following schedule:

> . On the first full week of the month, you have access to new date-patches of updated Mule runtimes.

Reviewer: ```suggestion
. On the first full week of the month, you have access to new date-patches of updated Mule runtime instances.
```

> . During the next two weeks, you can redeploy your applications using the newer date-patch to validate its behavior. This is known as a grace period.

Reviewer: ```suggestion
. During the next two weeks, you can redeploy your applications using the newer date-patch to validate its behavior. This is known as a _grace period_.
```

. Beginning on the third full full week of the month, you must select the latest date-patch of your chosen Mule runtime version when deploying a new application, or when making a change to an existing application.

[NOTE]

> During the grace period, you can select earlier date-patch versions until they are no longer available. If you select an earlier date-patch, and make any subsequent changes to that application, you may be required to update its date-patch version to a later release.

Reviewer: ```suggestion
During the grace period, you can select earlier date-patch versions until they are no longer available. If you select an earlier date-patch, and make any subsequent changes to that application, you might be required to update its date-patch version to a later release.
```
"until they are no longer available" = ? deprecated? removed from <what>?

== Update a Single App to the Latest Runtime Patch Release

Before updating apps in the production environment, update them in the sandbox environment and run required tests. 

> To update an app to the latest date patch:

Reviewer: Given the previous statement, is this and what follows AFTER the sandbox update and test?

> . In the Runtime Manager left menu, click *Applications*.

Reviewer: Is the "left menu" the "navigation menu"?

> . In the Runtime Manager left menu, click *Applications*.

Author: ```suggestion
. In the Runtime Manager navigation menu, click *Applications*.
```

> . Under *Deployment Target*, from the *Runtime version* list, select the version tagged `new`.

Reviewer: colon pointing to image
```suggestion
. Under *Deployment Target*, from the *Runtime version* list, select the version tagged `new`:
```

+
image::rtf-runtime-select-version.png[The new runtime version is selected]
Although the version number of the update might be the same as the installed version, a later date identifies the update:
+
image::rtf-runtime-select-latest-version.png[The latest runtime version indicated by the date]
. If you want to see details about the version, click the *Read release notes* link.

> . Click the *Apply Changes* button to update the app. 

Reviewer: Just sayin', here and in most other cases, you can omit identifying the UI control type:
```suggestion
. Click *Apply Changes* to update the app. 
```

+
Runtime Fabric updates the app with the version you selected and displays a message indicating success or failure.

> . Re-deploy your application.

Reviewer: ```suggestion
. Redeploy your application.
```

+

> Mule deployments on Runtime Fabric are not automatically patched. You must re-deploy your applications to apply the latest date-patch updates.

Reviewer: ```suggestion
Mule deployments on Runtime Fabric are not automatically patched. You must redeploy your applications to apply the latest date-patch updates.
```

= Mule Runtime Patch Updates for Runtime Fabric 

The following provides important information about Mule runtime engine patch updates for Runtime Fabric.

== Monthly Date-Patch Updates 

Once per month, MuleSoft releases a _date-patch update_, which includes backward-compatible bug fixes to address any issues discovered in Mule runtime engine.
The monthly date-patch update might include security patches if the scheduled date is within the security SLA.

The format of the version number for these updates is:

_Major_._Minor_._Patch_ - _Date_ 

Here's an example version number, including the patch date:

`4.3.0:20210101`

> The monthly patches update applications only to the latest date patch for the patch version.

Reviewer 2: I'm confused by this sentence.

> The monthly patches update applications only to the latest date patch for the patch version.

Author: ```suggestion
The monthly patches update applications only to the latest date-patch. They do not change the runtime version.
```

> They do not change the runtime version.

Author: ```suggestion
```

== Runtime Fabric Runtime Patch Update Schedule 

Runtime Fabric runs each application and gateway deployment on a separate Mule runtime instance. By managing the containerized Mule images used for these deployments, Runtime Fabric enables you to upgrade your Mule runtime version for each deployment on the following schedule:

> . On the first full week of the month, you have access to new versions of updated Mule runtimes.

Reviewer 2: suggest: "to new versions" --> ..to new date-patches"

> . On the first full week of the month, you have access to new versions of updated Mule runtimes.

Author: ```suggestion
. On the first full week of the month, you have access to new date-patches of updated Mule runtimes.
```

> . During the next two weeks, you can redeploy your applications using the newer runtime versions to validate its behavior. This is known as a grace period.

Reviewer 2: suggest: "runtime versions" --> ..date-patch"

> . During the next two weeks, you can redeploy your applications using the newer runtime versions to validate its behavior. This is known as a grace period.

Author: ```suggestion
. During the next two weeks, you can redeploy your applications using the newer date-patch to validate its behavior. This is known as a grace period.
```

> . Beginning on the third full full week of the month, you must select the latest date patch version of your chosen Mule runtime version when deploying a new application, or when making a change to an existing application.

Reviewer 2: suggest omitting "version" when discussing "date patch version"

Reviewer 2: suggest adding hyphen: "date patch" --> "date-patch"

> . Beginning on the third full full week of the month, you must select the latest date patch version of your chosen Mule runtime version when deploying a new application, or when making a change to an existing application.

Author: ```suggestion
. Beginning on the third full full week of the month, you must select the latest date-patch of your chosen Mule runtime version when deploying a new application, or when making a change to an existing application.
```

[NOTE]

> During the grace period, you can select earlier date-patch versions until they are no longer available. If you select an earlier date-patch version, and make any subsequent changes to that application, you may be required to update its date-patch version to a later release.

Reviewer 2: suggest: "can select an earlier date-patch" (omit "version" when used after "date-patch").

> During the grace period, you can select earlier date-patch versions until they are no longer available. If you select an earlier date-patch version, and make any subsequent changes to that application, you may be required to update its date-patch version to a later release.

Author: ```suggestion
During the grace period, you can select earlier date-patch versions until they are no longer available. If you select an earlier date-patch, and make any subsequent changes to that application, you may be required to update its date-patch version to a later release.
```

= Mule Runtime Patch Updates for Runtime Fabric 

The following provides important information about Mule runtime engine patch updates for Runtime Fabric.

== Monthly Date-Patch Updates 

> Once per month, MuleSoft releases a _date-patch update_,  which includes backward-compatible bug fixes to address any issues discovered in Mule runtime engine.

Author: ```suggestion
Once per month, MuleSoft releases a _date-patch update_, which includes backward-compatible bug fixes to address any issues discovered in Mule runtime engine.
```

The monthly date-patch update might include security patches if the scheduled date is within the security SLA.

The format of the version number for these updates is:

_Major_._Minor_._Patch_ - _Date_ 

Here's an example version number, including the patch date:

`4.3.0:20210101`

The monthly patches update applications only to the latest date patch for the patch version.

> They do not change the patch version number.

Reviewer 3: This is somewhat confusing. Perhaps we could say "They do not change the Runtime Version" instead?

Author: Ok, fixed.

> They do not change the patch version number.

Author: ```suggestion
They do not change the runtime version.
```

== Runtime Fabric Runtime Patch Update Schedule 

Runtime Fabric runs each application and gateway deployment on a separate Mule runtime instance. By managing the containerized Mule images used for these deployments, Runtime Fabric enables you to upgrade your Mule runtime version for each deployment on the following schedule:

. On the first full week of the month, you have access to new versions of updated Mule runtimes.
. During the next two weeks, you can redeploy your applications using the newer runtime versions to validate its behavior. This is known as a grace period.
. Beginning on the third full full week of the month, you must select the latest date patch version of your chosen Mule runtime version when deploying a new application, or when making a change to an existing application.

[NOTE]

> If needed, you can redeploy an application using an older runtime version. However, if you make any subsequent changes to that application, you must update its runtime version. 

Author: ```suggestion
If needed, you can redeploy an application using an older runtime version. However, if you make any subsequent changes to that application, you must update its runtime version to the latest release. 
```

> If needed, you can redeploy an application using an older runtime version. However, if you make any subsequent changes to that application, you must update its runtime version to the latest release. 

Author: ```suggestion
During the grace period, you can select earlier date-patch versions until they are no longer available. If you select an earlier date-patch version, and make any subsequent changes to that application, you may be required to update its date-patch version to a later release.
```

== Update a Single App to the Latest Runtime Patch Release

Before updating apps in the production environment, update them in the sandbox environment and run required tests. 

To update an app to the latest date patch:

. In the Runtime Manager left menu, click *Applications*.

> . Under *Deployment Target*, from the *Runtime version* menu, select the new version.

Author: ```suggestion
. Under *Deployment Target*, from the *Runtime version* menu, select the version tagged `new`.
```

> . Under *Deployment Target*, from the *Runtime version* menu, select the version tagged `new`.

Author: ```suggestion
. Under *Deployment Target*, from the *Runtime version* list, select the version tagged `new`.
```

+
image::rtf-runtime-select-version.png[The new runtime version is selected]
Although the version number of the update might be the same as the installed version, a later date identifies the update:
+
image::rtf-runtime-select-latest-version.png[The latest runtime version indicated by the date]
. If you want to see details about the version, click the *Read release notes* link.
. Click the *Apply Changes* button to update the app. 
+
Runtime Fabric updates the app with the version you selected and displays a message indicating success or failure.
. Re-deploy your application.
+
Mule deployments on Runtime Fabric are not automatically patched. You must re-deploy your applications to apply the latest date-patch updates.

== See Also

> * Link to Best Practices with Mule Patching[]

Author: ```suggestion
```
